### ISSUE TITLE
"build -f" fails when ADD-ing while in symlinked directory

### ISSUE THREAD

**Issue by oeuftete** [docker/docker#15642]
Here's a minimal example with docker 1.8.1.  Although there are maybe some hints of this in Issue #14339, I think this should narrow down at least one specific problem case.  I couldn't find this elsewhere.

```text
ken@oblako:/tmp $ mkdir /tmp/foo
ken@oblako:/tmp $ ln -s /tmp/foo /tmp/bar
ken@oblako:/tmp $ cd /tmp/bar
ken@oblako:/tmp/bar $ touch quux
ken@oblako:/tmp/bar $ echo "FROM scratch" > dock_er_file 
ken@oblako:/tmp/bar $ echo "ADD quux /" >> dock_er_file
ken@oblako:/tmp/bar $ docker build -f dock_er_file .
unable to prepare context: The Dockerfile (/tmp/bar/dock_er_file) must be within the build context (.)
ken@oblako:/tmp/bar $ cd /tmp/foo
ken@oblako:/tmp/foo $ docker build -f dock_er_file .
Sending build context to Docker daemon 3.584 kB
Step 0 : FROM scratch
 ---> 
Step 1 : ADD quux /
 ---> Using cache
 ---> d98f184b7fe1
Successfully built d98f184b7fe1
ken@oblako:/tmp/foo $ docker --version
Docker version 1.8.1, build d12ea79
ken@oblako:/tmp/foo $ docker info
Containers: 0
Images: 310
Storage Driver: aufs
 Root Dir: /media/r0/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 330
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-61-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 8
Total Memory: 15.57 GiB
[ ... some redaction ... ]
WARNING: No swap limit support
ken@oblako:/tmp/foo $ uname -a
Linux oblako 3.13.0-61-generic #100-Ubuntu SMP Wed Jul 29 11:21:34 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by tonistiigi** [comment]
This was already fixed in master in #15060.

**Comment by thaJeztah** [comment]
Ah, looks like it yes. Thanks @tonistiigi 

I'm closing this issue, because this should be resolved in https://github.com/docker/docker/pull/15060
